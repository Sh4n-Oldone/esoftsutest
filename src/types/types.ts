export interface ImageData {
  id: string;
  url: string;
  urlBiggest: string;
  description: string;
  created: string;
  author: string;
  tags: string[];
}

export interface ImageFullData extends ImageData {
  alt_description: string;
  topic_submissions: object[];
  created_at: string;
  user: {
    username: string;
  };
  urls: {
    regular: string;
    full: string;
  };
}
