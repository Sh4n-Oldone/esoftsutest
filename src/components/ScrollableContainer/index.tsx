import React from 'react';
import {
  Image,
  ScrollView,
  StyleSheet,
  TouchableOpacity,
  View,
} from 'react-native';
import {ImageData} from '../../types/types';

const ScrollableContainer: React.FC<{
  componentWidth?: string;
  images: ImageData[];
  imageClick: (image: ImageData) => void;
  newContainerStyles?: {}[];
  newImageStyles?: {}[];
}> = ({
  componentWidth,
  images,
  imageClick,
  newContainerStyles,
  newImageStyles,
}) => {
  return (
    <ScrollView style={{width: componentWidth || '100%'}}>
      <View style={newContainerStyles || [styles.imagesContainer]}>
        {images.map((image, index: number) => (
          <TouchableOpacity
            key={image.id}
            style={
              newImageStyles || [
                styles.imageContainer,
                index === images.length - 1 && {marginBottom: 0},
              ]
            }
            onPress={() => imageClick(image)}>
            <Image
              key={image.id}
              source={{uri: image.url}}
              style={styles.image}
            />
          </TouchableOpacity>
        ))}
      </View>
    </ScrollView>
  );
};

const styles = StyleSheet.create({
  imagesContainer: {
    paddingBottom: 46,
  },
  imageContainer: {
    height: 400,
    marginBottom: 10,
  },
  image: {
    height: '100%',
    width: '100%',
  },
});

export default ScrollableContainer;
