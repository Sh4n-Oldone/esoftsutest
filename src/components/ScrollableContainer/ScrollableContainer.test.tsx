import 'react-native';
import React from 'react';
import ScrollableContainer from './index';
import renderer from 'react-test-renderer';
import {ImageData} from '../../types/types';

const dataPlug: ImageData[] = [
  {
    id: 'jjhvyxm34nY',
    created: '2017-09-01',
    description: 'Coworking space Industrious Atlanta',
    urlBiggest: 'https://images.unsplash.com/photo-1504297050567',
    url: 'https://images.unsplash.com/photo-1504297050568-910d24c426d3',
    tags: ['business-work', 'interiors'],
    author: 'Mia Baker',
  },
  {
    id: 'jjhvyxm',
    created: '2017-09-02',
    description: 'cattos',
    urlBiggest: 'https://images.unsplash.com/photo-1504297050568',
    url: 'https://images.unsplash.com/photo-1504297050568-910d24c426d4',
    tags: ['cats', 'home'],
    author: 'Mia Alb',
  },
];

describe('Correctly rendered ScrollableContainer', () => {
  it('Rendered with necessary props', () => {
    const container = renderer.create(
      <ScrollableContainer imageClick={() => {}} images={dataPlug} />,
    );
    expect(container.root.findAllByType('Image')[0].props.source).toStrictEqual(
      {
        uri: 'https://images.unsplash.com/photo-1504297050568-910d24c426d3',
      },
    );
    expect(container.root.findAllByType('Image')[1].props.source).toStrictEqual(
      {
        uri: 'https://images.unsplash.com/photo-1504297050568-910d24c426d4',
      },
    );
    expect(container.toJSON()).toMatchSnapshot();
  });

  it('Rendered with all props', () => {
    const container = renderer.create(
      <ScrollableContainer
        imageClick={() => {}}
        images={dataPlug}
        componentWidth={'200'}
        newContainerStyles={[{backGroundColor: '#000'}]}
        newImageStyles={[{backGroundColor: '#fff'}]}
      />,
    );
    expect(container.toJSON()).toMatchSnapshot();
    expect(container.root.findAllByType('Image')[0].props.source).toStrictEqual(
      {
        uri: 'https://images.unsplash.com/photo-1504297050568-910d24c426d3',
      },
    );
    expect(container.root.findAllByType('Image')[1].props.source).toStrictEqual(
      {
        uri: 'https://images.unsplash.com/photo-1504297050568-910d24c426d4',
      },
    );
    
    // и ScrollView, и TouchableOpacity наследуются от View,
    expect(container.root.findAllByType('View')[1].props.style).toStrictEqual([
      {backGroundColor: '#000'},
    ]);
    expect(
      container.root.findAllByType('View')[2].props.style.backGroundColor,
    ).toBe('#fff');
    expect(container.toJSON().props.style.width).toBe('200');
  });
});
