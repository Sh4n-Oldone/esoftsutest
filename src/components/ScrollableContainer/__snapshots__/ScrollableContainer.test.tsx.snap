// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Correctly rendered ScrollableContainer Rendered with all props 1`] = `
<RCTScrollView
  style={
    Object {
      "width": "200",
    }
  }
>
  <View>
    <View
      style={
        Array [
          Object {
            "backGroundColor": "#000",
          },
        ]
      }
    >
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        nativeID="animatedComponent"
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "backGroundColor": "#fff",
            "opacity": 1,
          }
        }
      >
        <Image
          source={
            Object {
              "uri": "https://images.unsplash.com/photo-1504297050568-910d24c426d3",
            }
          }
          style={
            Object {
              "height": "100%",
              "width": "100%",
            }
          }
        />
      </View>
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        nativeID="animatedComponent"
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "backGroundColor": "#fff",
            "opacity": 1,
          }
        }
      >
        <Image
          source={
            Object {
              "uri": "https://images.unsplash.com/photo-1504297050568-910d24c426d4",
            }
          }
          style={
            Object {
              "height": "100%",
              "width": "100%",
            }
          }
        />
      </View>
    </View>
  </View>
</RCTScrollView>
`;

exports[`Correctly rendered ScrollableContainer Rendered with necessary props 1`] = `
<RCTScrollView
  style={
    Object {
      "width": "100%",
    }
  }
>
  <View>
    <View
      style={
        Array [
          Object {
            "paddingBottom": 46,
          },
        ]
      }
    >
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        nativeID="animatedComponent"
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "height": 400,
            "marginBottom": 10,
            "opacity": 1,
          }
        }
      >
        <Image
          source={
            Object {
              "uri": "https://images.unsplash.com/photo-1504297050568-910d24c426d3",
            }
          }
          style={
            Object {
              "height": "100%",
              "width": "100%",
            }
          }
        />
      </View>
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        nativeID="animatedComponent"
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "height": 400,
            "marginBottom": 0,
            "opacity": 1,
          }
        }
      >
        <Image
          source={
            Object {
              "uri": "https://images.unsplash.com/photo-1504297050568-910d24c426d4",
            }
          }
          style={
            Object {
              "height": "100%",
              "width": "100%",
            }
          }
        />
      </View>
    </View>
  </View>
</RCTScrollView>
`;
